File: mobile/src/screens/auth/LoginScreen.js
import { useState } from 'react';
import { View, Text, StyleSheet, Pressable, KeyboardAvoidingView, Platform, ScrollView } from 'react-native';
import Screen from '../../components/common/Screen';
import Input from '../../components/common/Input';
import Button from '../../components/common/Button';
import { useAuth } from '../../hooks/useAuth';
import { colors, radius } from '../../config/theme';

function PasswordLogin() {
  const { login } = useAuth();
  const [identifier, setIdentifier] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');
  const [loading, setLoading] = useState(false);

  const submit = async () => {
    setError('');
    setLoading(true);
    try {
      await login({ identifier, password });
    } catch (err) {
      setError(err.response?.data?.message || 'Invalid credentials');
    } finally {
      setLoading(false);
    }
  };

  return (
    <View style={styles.form}>
      <Input placeholder="you@example.com" autoCapitalize="none" value={identifier} onChangeText={setIdentifier} label="Email or Username" />
      <Input placeholder="Password" secureTextEntry value={password} onChangeText={setPassword} label="Passphrase" />
      {error ? <Text style={styles.errorText}>{error}</Text> : null}
      <Button title={loading ? 'Signing in...' : 'Sign In to VartaKaro'} onPress={submit} loading={loading} />
    </View>
  );
}

function OtpLogin() {
  const { requestOtp, verifyOtp } = useAuth();
  const [phone, setPhone] = useState('');
  const [code, setCode] = useState('');
  const [sent, setSent] = useState(false);
  const [error, setError] = useState('');
  const [loading, setLoading] = useState(false);

  const send = async () => {
    setError('');
    setLoading(true);
    try {
      await requestOtp(phone);
      setSent(true);
    } catch (err) {
      setError(err.response?.data?.message || 'Could not send code');
    } finally {
      setLoading(false);
    }
  };

  const verify = async () => {
    setError('');
    setLoading(true);
    try {
      await verifyOtp(phone, code);
    } catch (err) {
      setError(err.response?.data?.message || 'Invalid code');
    } finally {
      setLoading(false);
    }
  };

  if (!sent) {
    return (
      <View style={styles.form}>
        <Input placeholder="+91 98765 43210" keyboardType="phone-pad" value={phone} onChangeText={setPhone} label="Phone number" />
        {error ? <Text style={styles.errorText}>{error}</Text> : null}
        <Button title={loading ? 'Sending...' : 'Send code'} onPress={send} loading={loading} />
        <Text style={styles.hint}>Dev mode: the code is printed in the server console.</Text>
      </View>
    );
  }

  return (
    <View style={styles.form}>
      <Input placeholder="123456" keyboardType="number-pad" value={code} onChangeText={setCode} label="6-digit code" />
      {error ? <Text style={styles.errorText}>{error}</Text> : null}
      <Button title={loading ? 'Verifying...' : 'Verify & continue'} onPress={verify} loading={loading} />
    </View>
  );
}

export default function LoginScreen({ navigation }) {
  const [tab, setTab] = useState('password');

  return (
    <Screen>
      <KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : undefined} style={{ flex: 1 }}>
        <ScrollView contentContainerStyle={styles.scroll} keyboardShouldPersistTaps="handled">
          <View style={styles.brandRow}>
            <View style={styles.brandBadge} />
            <View>
              <Text style={styles.brandName}>VartaKaro</Text>
              <Text style={styles.brandTagline}>Stories &amp; Conversations</Text>
            </View>
          </View>

          <View style={styles.card}>
            <Text style={styles.eyebrow}>✦ WELCOME BACK</Text>
            <Text style={styles.headline}>Welcome Back to the Sanctuary</Text>
            <Text style={styles.subtext}>Enter your email or username and passphrase to access your chats and circles.</Text>

            <View style={styles.tabRow}>
              <Pressable style={[styles.tab, tab === 'password' && styles.tabActive]} onPress={() => setTab('password')}>
                <Text style={[styles.tabText, tab === 'password' && styles.tabTextActive]}>Password</Text>
              </Pressable>
              <Pressable style={[styles.tab, tab === 'otp' && styles.tabActive]} onPress={() => setTab('otp')}>
                <Text style={[styles.tabText, tab === 'otp' && styles.tabTextActive]}>Phone code</Text>
              </Pressable>
            </View>

            {tab === 'password' ? <PasswordLogin /> : <OtpLogin />}
          </View>

          <Pressable style={styles.footerLink} onPress={() => navigation.navigate('Register')}>
            <Text style={styles.footerText}>
              Don&apos;t have an account? <Text style={styles.footerAccent}>Create Account</Text>
            </Text>
          </Pressable>

          <View style={styles.privacyBox}>
            <Text style={styles.privacyTitle}>Privacy, simply done</Text>
            <Text style={styles.privacyBody}>No behavioral trackers, third-party analytics, or ad surveillance anywhere in VartaKaro.</Text>
          </View>
        </ScrollView>
      </KeyboardAvoidingView>
    </Screen>
  );
}

const styles = StyleSheet.create({
  scroll: { padding: 20, paddingTop: 40 },
  brandRow: { flexDirection: 'row', alignItems: 'center', marginBottom: 24, gap: 10 },
  brandBadge: { width: 32, height: 32, borderRadius: radius.sm, backgroundColor: colors.accent },
  brandName: { fontWeight: '800', color: colors.accent, fontSize: 16 },
  brandTagline: { fontSize: 11, color: colors.inkSoft },
  card: { backgroundColor: colors.paper, borderRadius: radius.xl, padding: 20 },
  eyebrow: { color: colors.accent, fontWeight: '700', fontSize: 12, marginBottom: 8, letterSpacing: 0.5 },
  headline: { fontSize: 24, fontWeight: '800', color: colors.ink, marginBottom: 8 },
  subtext: { color: colors.inkSoft, marginBottom: 20, lineHeight: 20 },
  tabRow: { flexDirection: 'row', backgroundColor: colors.paperSoft, borderRadius: radius.full, padding: 4, marginBottom: 18, gap: 4 },
  tab: { flex: 1, paddingVertical: 8, borderRadius: radius.full, alignItems: 'center' },
  tabActive: { backgroundColor: colors.paper },
  tabText: { color: colors.inkSoft, fontWeight: '600', fontSize: 13 },
  tabTextActive: { color: colors.accent },
  form: { gap: 14 },
  errorText: { color: colors.danger, fontSize: 13 },
  hint: { fontSize: 12, color: colors.inkSoft },
  footerLink: { alignItems: 'center', marginTop: 18 },
  footerText: { color: colors.inkSoft, fontSize: 14 },
  footerAccent: { color: colors.accent, fontWeight: '700' },
  privacyBox: { backgroundColor: colors.paper, borderRadius: radius.lg, padding: 16, marginTop: 16 },
  privacyTitle: { fontWeight: '700', marginBottom: 4 },
  privacyBody: { color: colors.inkSoft, fontSize: 13, lineHeight: 18 },
});

File: mobile/lib/features/auth/screens/login_screen.dart
import 'package:dio/dio.dart';
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';

import '../../../common/widgets/app_button.dart';
import '../../../common/widgets/app_input.dart';
import '../../../common/widgets/screen.dart';
import '../../../config/theme.dart';
import '../providers/auth_provider.dart';

String errorMessage(Object err) {
  if (err is DioException) {
    final data = err.response?.data;
    if (data is Map && data['message'] is String) {
      return data['message'] as String;
    }
  }
  return 'Something went wrong. Please try again.';
}

class LoginScreen extends StatefulWidget {
  const LoginScreen({super.key});

  @override
  State<LoginScreen> createState() => _LoginScreenState();
}

class _LoginScreenState extends State<LoginScreen>
    with SingleTickerProviderStateMixin {
  late final TabController _tabController = TabController(
    length: 2,
    vsync: this,
  );

  @override
  void dispose() {
    _tabController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Screen(
      child: Scaffold(
        backgroundColor: Colors.transparent,
        body: Column(
          children: [
            const SizedBox(height: 32),
            const Text(
              'Vartakaro',
              style: TextStyle(
                fontSize: 28,
                fontWeight: FontWeight.w700,
                color: AppColors.ink,
              ),
            ),
            const SizedBox(height: 24),
            TabBar(
              controller: _tabController,
              labelColor: AppColors.accent,
              unselectedLabelColor: AppColors.inkSoft,
              indicatorColor: AppColors.accent,
              tabs: const [
                Tab(text: 'Password'),
                Tab(text: 'Phone code'),
              ],
            ),
            Expanded(
              child: TabBarView(
                controller: _tabController,
                children: const [_PasswordLogin(), _OtpLogin()],
              ),
            ),
            Padding(
              padding: const EdgeInsets.fromLTRB(20, 20, 20, 4),
              child: TextButton(
                onPressed: () => context.go('/register'),
                child: const Text(
                  "Don't have an account? Register",
                  style: TextStyle(color: AppColors.accent),
                ),
              ),
            ),
            Padding(
              padding: const EdgeInsets.fromLTRB(20, 0, 20, 20),
              child: TextButton(
                onPressed: () => context.push('/random-chat'),
                child: const Text(
                  'Talk to a stranger — no account needed',
                  style: TextStyle(color: AppColors.inkSoft),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class _PasswordLogin extends ConsumerStatefulWidget {
  const _PasswordLogin();

  @override
  ConsumerState<_PasswordLogin> createState() => _PasswordLoginState();
}

class _PasswordLoginState extends ConsumerState<_PasswordLogin> {
  final _identifier = TextEditingController();
  final _password = TextEditingController();
  bool _loading = false;
  String? _error;

  @override
  void dispose() {
    _identifier.dispose();
    _password.dispose();
    super.dispose();
  }

  Future<void> _submit() async {
    setState(() {
      _loading = true;
      _error = null;
    });
    try {
      await ref
          .read(authActionsProvider)
          .login(identifier: _identifier.text.trim(), password: _password.text);
    } catch (err) {
      setState(() => _error = errorMessage(err));
    } finally {
      if (mounted) setState(() => _loading = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    return SingleChildScrollView(
      padding: const EdgeInsets.all(20),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          AppInput(
            label: 'Username or email',
            controller: _identifier,
            hintText: 'you@example.com',
          ),
          const SizedBox(height: 16),
          AppInput(
            label: 'Password',
            controller: _password,
            obscureText: true,
            hintText: '••••••••',
          ),
          if (_error != null) ...[
            const SizedBox(height: 12),
            Text(
              _error!,
              style: const TextStyle(color: AppColors.danger, fontSize: 13),
            ),
          ],
          const SizedBox(height: 20),
          AppButton(
            label: 'Log in',
            onPressed: _loading ? null : _submit,
            loading: _loading,
          ),
        ],
      ),
    );
  }
}

class _OtpLogin extends ConsumerStatefulWidget {
  const _OtpLogin();

  @override
  ConsumerState<_OtpLogin> createState() => _OtpLoginState();
}

class _OtpLoginState extends ConsumerState<_OtpLogin> {
  final _phone = TextEditingController();
  final _code = TextEditingController();
  bool _sent = false;
  bool _loading = false;
  String? _error;

  @override
  void dispose() {
    _phone.dispose();
    _code.dispose();
    super.dispose();
  }

  Future<void> _requestCode() async {
    setState(() {
      _loading = true;
      _error = null;
    });
    try {
      await ref.read(authActionsProvider).requestOtp(_phone.text.trim());
      setState(() => _sent = true);
    } catch (err) {
      setState(() => _error = errorMessage(err));
    } finally {
      if (mounted) setState(() => _loading = false);
    }
  }

  Future<void> _verify() async {
    setState(() {
      _loading = true;
      _error = null;
    });
    try {
      await ref
          .read(authActionsProvider)
          .verifyOtp(phone: _phone.text.trim(), code: _code.text.trim());
    } catch (err) {
      setState(() => _error = errorMessage(err));
    } finally {
      if (mounted) setState(() => _loading = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    return SingleChildScrollView(
      padding: const EdgeInsets.all(20),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          AppInput(
            label: 'Phone number',
            controller: _phone,
            hintText: '+1 555 0100',
            keyboardType: TextInputType.phone,
          ),
          if (_sent) ...[
            const SizedBox(height: 8),
            const Text(
              'Dev mode: the code is printed in the server console.',
              style: TextStyle(color: AppColors.inkSoft, fontSize: 12),
            ),
            const SizedBox(height: 16),
            AppInput(
              label: '6-digit code',
              controller: _code,
              keyboardType: TextInputType.number,
            ),
          ],
          if (_error != null) ...[
            const SizedBox(height: 12),
            Text(
              _error!,
              style: const TextStyle(color: AppColors.danger, fontSize: 13),
            ),
          ],
          const SizedBox(height: 20),
          AppButton(
            label: _sent ? 'Verify code' : 'Send code',
            onPressed: _loading ? null : (_sent ? _verify : _requestCode),
            loading: _loading,
          ),
        ],
      ),
    );
  }
}
